project/src/molecules/system_builder.py

```diff
@@ -1,33 +1,65 @@
 """Methods used to create systems from job statepoint."""
 import mbuild as mb
-import signac
-from mbuild import Box
+from methane_ua import MethaneUA
+from pentane_ua import PentaneUA
 
-from project.src import molecules
 
-from .methane_ua import MethaneUA
+def construct_system(sp):
+    """Construct systems according to job statepoint.
 
+    Parameters
+    ----------
+    sp: dict (from job.sp)
+        Dictionary contains information necessary to construct a system.
+        Stored as state of job. The dictionary should resemble:
+        {"molecule": str,
+         "engine": str,
+         "replica": int,
+         "temperature": float (in K),
+         "pressure": float (in kPa),
+         "ensemble": str,
+         "N_liquid": int,
+         "N_vap": int,
+         "box_L_liq": int (nm),
+         "box_L_vap", int (nm),
+         "init_liq_den": float (g/cm3),
+         "init_vap_den": float (g/cm3),
+         "mass": float (g/mol),
+         "forcefield_name": str,
+         "cutoff_style": str,
+         "r_cut": float (in nm)}
 
-def SystemBuilder(job):
-    """Construct systems according to job statepoint."""
-    # Update this dict as new reicpes are made
+    Returns
+    -------
+    system: mb.Compound
+        The constructed mbuild Compound
+    """
+    system = mb.Compound()
+
+    # Update this dict as new recipes are made
     molecule_dict = {
         "methaneUA": MethaneUA,
-        "pentaneUA": None,
+        "pentaneUA": PentaneUA,
         "benzeneUA": None,
         "waterSPC/E": None,
         "ethanolAA": None,
     }
-    molecule = molecule_dict[job.sp["molecule"]]
-    vap_box = mb.Box()
-    liq_box = mb.Box()
-
-    vap_box_cpd = mb.fill_box(
-        compound=molecule, n_compound=job.sp["n_compounds"], box=vap_box
-    )
-    liq_box_cpd = mb.fill_box(
-        compound=molecule, n_compound=job.sp["n_compounds"], box=liq_box
+    molecule = molecule_dict[sp["molecule"]]
+    liq_box = mb.Box([sp["box_L_liq"]] * 3)
+    filled_liq_box = mb.fill_box(
+        compound=[molecule], n_compounds=[sp["N_liquid"]], box=liq_box
     )
+    system.add(liq_box)
+
+    if sp["box_L_vap"] and sp["N_vap"]:
+        vap_box = mb.Box([sp["box_L_vap"]] * 3)
+        filled_vap_box = mb.fill_box(
+            compound=[molecule], n_compounds=[sp["N_vap"]], box=vap_box
+        )
+
+        # Translate the vapor box to be on top of the liquid box
```

Comment:
We need to verify with @ramanishsingh @rwsmith7531 and @bc118  if this is how they need the system to be set up. They might need two separate boxes anyway for their simulations.

Reply:
Yes, I need some input about how the two box need to be stacked. Also, the individual box objects should still be accessible from the system (e.g., `system.children[0].box`)

Reply:
Can we also return the molecule? Something like

```python
       return molecule, system
```
Or will it be messy?

Otherwise I can get the molecule from the recipe.

Reply:
Also, right now the system is an mBuild compound. 

Instead, can we make a list of boxes? 

```python
    box_list=[]

    boxes.append(filled_liq_box)
    
    boxes.append(filled_vap_box)
    
    return box_list
```

And then we can access box_list[1] when we are in GEMC. 
I think that way it will be simpler if we keep the two boxes separate.

Reply:
this is doable but it will make the output kinda messy (and may cause confusion for engines that do not need these info). I think these information can technically be retrieved from the `system` object without too much difficulty. 

Getting molecule info:
```
particle = system[1] # This is valid in here since we only have one kind of molecule
```

Getting list of boxes 
```
box_list = [child.box for child in system.children]
```

Reply:
I think @ramanishsingh means being able to access the vap and liq box as a list of two compounds instead of one. 

So for cases where N_vap or any of the vap statepoint items are None, it’s just a None in the location where the filled vap box would be. 

I think the main thing is that each of these compounds (liq and vap) do not have to be thought of as a single compound, but two compounds? @ramanishsingh is that correct?

Reply:
Or if we add labels for each compound in the system “liquid” and “vap” or something when adding them together could also work I think.

Reply:
"I think the main thing is that each of these compounds (liq and vap) do not have to be thought of as a single compound, but two compounds? @ramanishsingh is that correct?"

Yes!

Reply:
ok, I see. Then, how about we will return a single filled liquid box if the the vap box is None. If both of them exist, we will return a list of compound ([liq, vap]).

Reply:
actually, would you prefer a list or a dictionary of system (i.e., `[liq_box, vap_box]` vs `{"liquid": liq_box, "vapor": vap_box}`?

Reply:
Fix included in the latest commit (right now I am returning a list of boxes, but lmk if you prefer a dictionary)

Reply:
Thanks!